java-squid/src/main/java/org/sonar/java/se/ExplodedGraph.java

```diff
@@ -84,6 +84,7 @@ public boolean equals(Object obj) {
     final ProgramPoint programPoint;
     @Nullable
     final ProgramState programState;
+    public boolean exitPath = false;
```

Comment:
I think this is the way to go: the program state should memorize the mode of executing a finally block (with successors or exit path)